src/lighteval/models/endpoints/inference_providers_model.py

```diff
@@ -0,0 +1,273 @@
+        prompts,
+        num_samples: int | list[int],
+    ):
+        results = []
+
+        num_sampless = [num_samples for _ in prompts] if not isinstance(num_samples, list) else num_samples
+        assert (
+            len(prompts) == len(num_sampless)
+        ), f"Length of prompts, return_logitss, max_new_tokenss, num_sampless, stop_sequences, system_prompts should be the same but are {len(prompts)}, {len(num_sampless)}"
+
+        async def bounded_api_call(prompt, num_samples):
+            async with self.semaphore:
+                return await self.__call_api(prompt, num_samples)
+
+        tasks = [bounded_api_call(prompt, num_samples) for prompt, num_samples in zip(prompts, num_sampless)]
+        results = await async_tqdm.gather(*tasks)
+
+        if None in results:
+            raise ValueError("Some entries are not annotated due to errors in annotate_p, please inspect and retry.")
+
+        return results
+
+    def greedy_until(
+        self,
+        requests: list[GreedyUntilRequest],
+        override_bs: Optional[int] = None,
+    ) -> list[GenerativeResponse]:
+        """
+        Generates responses using a greedy decoding strategy until certain ending conditions are met.
+
+        Args:
+            requests (list[Request]): list of requests containing the context and ending conditions.
+            override_bs (int, optional): Override the batch size for generation. Defaults to None.
+
+        Returns:
+            list[GenerativeResponse]: list of generated responses.
+        """
+        for request in requests:
+            request.tokenized_context = self.tok_encode(request.context)
+
+        dataset = GenerativeTaskDataset(requests=requests, num_dataset_splits=self.DATASET_SPLITS)
+        results = []
+
+        for _ in tqdm(
+            dataset.splits_start_end_iterator(),
+            total=dataset.num_dataset_splits,
+            desc="Splits",
+            position=0,
+            disable=False,  # self.disable_tqdm,
+        ):
+            contexts = [c.context for c in dataset]
+            num_samples = dataset[0].num_samples
+
+            responses = asyncio.run(self.__call_api_parallel(contexts, num_samples))
+
+            for response in responses:
+                result: list[str] = [choice.message.content for choice in response.choices]
+
+                cur_response = GenerativeResponse(
+                    # In empty responses, the model should return an empty string instead of None
+                    result=result if result[0] else [""],
+                    logits=None,
+                    generated_tokens=[],
+                    input_tokens=[],
+                )
+                results.append(cur_response)
+
+        return dataset.get_original_order(results)
+
+    @property
+    def tokenizer(self):
+        return self._tokenizer
+
+    @property
+    def add_special_tokens(self) -> bool:
+        return False
+
+    @property
+    def max_length(self) -> int:
```

Comment:
Why should it be 4096?

Reply:
(since the models are going to be on the hub, we could preload them to extract the info)

Reply:
(I agree it will be a bad idea for the bigger models - maybe only loading the tk config?)